xbmc/pvr/PVRGUIInfo.cpp

```diff
@@ -853,52 +810,134 @@ bool CPVRGUIInfo::GetVideoLabel(const CFileItem *item, int iLabel, std::string &
   return false;
 }
 
-bool CPVRGUIInfo::GetSeekTimeLabel(int iSeekSize, std::string &strValue) const
+bool CPVRGUIInfo::GetMultiInfoLabel(const CFileItem *item, const GUIInfo &info, std::string &strValue) const
 {
-  strValue = StringUtils::SecondsToTimeString(GetElapsedTime() + iSeekSize, TIME_FORMAT_GUESS).c_str();
+  CSingleLock lock(m_critSection);
+
+  switch (info.m_info)
+  {
+    case PVR_EPG_EVENT_DURATION:
+      CharInfoEpgEventDuration(item, static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_EPG_EVENT_ELAPSED_TIME:
+      CharInfoEpgEventElapsedTime(item, static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_EPG_EVENT_REMAINING_TIME:
+      CharInfoEpgEventRemainingTime(item, static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_EPG_EVENT_FINISH_TIME:
+      CharInfoEpgEventFinishTime(item, static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_TIMESHIFT_START_TIME:
+      CharInfoTimeshiftStartTime(static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_TIMESHIFT_END_TIME:
+      CharInfoTimeshiftEndTime(static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_TIMESHIFT_PLAY_TIME:
+      CharInfoTimeshiftPlayTime(static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    case PVR_TIMESHIFT_OFFSET:
+      CharInfoTimeshiftOffset(static_cast<TIME_FORMAT>(info.GetData1()), strValue);
+      break;
+    default:
+      return false;
+  }
   return true;
```

Comment:
It seems this line can't be reached anymore

Reply:
It can. You see see `break;` in line 842, just as an example.